main/src/main/res/layout/activity_launch.xml

```diff
@@ -42,7 +42,7 @@
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/line_ads"
+        app:layout_constraintTop_toTopOf="@+id/admobAd"
```

Comment:
```suggestion
        app:layout_constraintTop_toTopOf="parent"
```
Please let it aligns the parent to make it not relative to any other views.

Reply:
And also, maybe we can remove the following block in this file since it is not used now.

```
    <androidx.constraintlayout.widget.Barrier
        android:id="@+id/line_ads"
        android:layout_width="0dp"
        android:layout_height="0dp"
        app:barrierDirection="bottom"
        app:constraint_referenced_ids="admobAd" />
```